packages/functions/src/index.ts

```diff
@@ -3,8 +3,53 @@ import * as functions from "firebase-functions";
 import {checkAuthToken} from "./middlewares/check-auth-token";
 import authRouter from "./resources/auth/auth.router";
 import userRouter from "./resources/user/user.router";
+import * as admin from "firebase-admin";
+import {Change, onDocumentCreated,
+  onDocumentDeleted,
+  onDocumentUpdated,
+  QueryDocumentSnapshot} from "firebase-functions/v2/firestore";
+import {updatePublishedEvent}
+  from "./resources/bot/actions/updatePublishedEvent";
+import {sendToChannel} from "./resources/bot/actions/sendToChannel";
+import {deleteTelegramMessage}
+  from "./resources/bot/actions/deleteTelegramMessage";
+
+admin.initializeApp();
+
+export const publishToTelegram =
+onDocumentCreated("events/{eventId}", (event) => {
+  const snapshot = event.data;
+  if (snapshot) {
```

Comment:
if post is in draft status, it shouldn't be sent to channel

Reply:
It doesn't. It cheks for status and messageId  
![image](https://github.com/user-attachments/assets/3d1fd5be-255f-4e0b-815d-4073c867da4b)